### DISCUSSION TITLE
Using Actix with AWS Lambda

### DISCUSSION THREAD

**Discussion by spirrello** [actix/actix-web#2841] [Q&A]
Hi,

I'm working on setting up an API with AWS Lambda + AWS API Gateway and would like to use the Actix HTTP router.  Is this possible?  If so can you provide an example?

Thank you for your help in advance.

- Stefano

**Comment by robjtede** [comment]
I've no experience in trying but I highly doubt that it's possible yet.

**Comment by vincent-thomas** [comment]
@spirrello Api Gateway is the router in that scenario, don't think api gateway has a rust runtime

**Comment by Samielakkad** [comment]
Using the full Actix Web server inside AWS Lambda/API Gateway is not the natural fit because Lambda does not give Actix a normal long-running socket listener. API Gateway receives the HTTP request and invokes your Lambda handler.

The usual Rust Lambda shape is:

```text
API Gateway event -> Lambda runtime handler -> your routing logic -> Lambda response
```

If you want Actix-like routing, use a framework-independent router crate such as `matchit` inside the Lambda handler. Example shape:

```rust
let mut router = matchit::Router::new();
router.insert("/users/:id", Route::UserShow).unwrap();
router.insert("/health", Route::Health).unwrap();

match router.at(path) {
    Ok(matched) => match matched.value {
        Route::UserShow => { /* build Lambda response */ }
        Route::Health => { /* build Lambda response */ }
    },
    Err(_) => { /* 404 */ }
}
```

If you want a full web framework on Lambda, look for frameworks/adapters designed around Lambda's event model rather than a socket server. Actix Web is strongest when it owns the HTTP server process directly, for example on ECS, EC2, Fly.io, a VM, or a container behind a load balancer.

**Comment by spirrello** [comment]
Thx for the matchit suggestion.  I'll take a look at that.
